### PR TITLE
Update go2rtc to 1.7.1

### PR DESCRIPTION
**Pull request by NickM-27**
This is a major update with a lot of changes, likely needs more testing before merging.

In my initial testing of the dev version over the last week: with a total of 12 stream consumers and 6 stream producers, CPU usage went from 12% of one core to 8% of one core when moving to the dev version

### CODE REVIEW

File: docs/docs/configuration/restream.md [hunk 1]
```diff
@@ -92,53 +92,69 @@ Two connections are made to the camera. One for the sub stream, one for the rest
...
+Restreaming to HomeKit is supported but there are some requirements:
+- Frigate must be run in host mode, this means that this feature is not supported in the addon
+- Only h264 video / opus audio is supported
+
+```yaml
+go2rtc:
+  streams:
+    your_camera: rtsp://192.168.1.5:554/live0
+  homekit:
+    your_camera:  # same stream ID from streams list, default PIN - 195502224
```
Comment (AlexxIT) [inline docs/docs/configuration/restream.md:150]: Wrong pin. 19550224

File: docker/main/Dockerfile [hunk 2]
```diff
@@ -33,7 +33,7 @@ RUN --mount=type=tmpfs,target=/tmp --mount=type=tmpfs,target=/var/cache/apt \
 FROM scratch AS go2rtc
 ARG TARGETARCH
 WORKDIR /rootfs/usr/local/go2rtc/bin
-ADD --link --chmod=755 "https://github.com/AlexxIT/go2rtc/releases/download/v1.6.2/go2rtc_linux_${TARGETARCH}" go2rtc
+ADD --link --chmod=755 "https://github.com/AlexxIT/go2rtc/releases/download/v1.7.0/go2rtc_linux_${TARGETARCH}" go2rtc
```
Comment (skrashevich) [inline docker/main/Dockerfile:36]:
```suggestion
ADD --link --chmod=755 "https://github.com/AlexxIT/go2rtc/releases/download/v1.7.1/go2rtc_linux_${TARGETARCH}" go2rtc
```

File: docs/docs/configuration/advanced.md [hunk 3]
```diff
@@ -120,7 +120,7 @@ NOTE: The folder that is mapped from the host needs to be the folder that contai
 
 ## Custom go2rtc version
 
-Frigate currently includes go2rtc v1.6.2, there may be certain cases where you want to run a different version of go2rtc.
+Frigate currently includes go2rtc v1.7.0, there may be certain cases where you want to run a different version of go2rtc.
```
Comment (skrashevich) [inline docs/docs/configuration/advanced.md:123]:
```suggestion
Frigate currently includes go2rtc v1.7.1, there may be certain cases where you want to run a different version of go2rtc.
```

File: docker/main/Dockerfile (hunk 2 above)
Comment (NickM-27) [reply docker/main/Dockerfile:36]: This is in draft, I changed the title but haven't pushed the code yet

**Review by blakeblackshear — APPROVED** [review]

**Pull request merged**